web-api/src/personalization/routes.rs

```diff
@@ -497,26 +526,34 @@ impl UnvalidatedSemanticSearchRequest {
             include_properties,
             filter,
         } = self;
-        let document = document.validate()?;
         let semantic_search_config: &SemanticSearchConfig = config.as_ref();
-        let count = validate_count(
+        let tenants_config: &tenants::Config = config.as_ref();
+
+        let document = document.validate()?;
+        let count = count.unwrap_or(semantic_search_config.default_number_documents);
+        dev.validate(tenants_config.enable_dev, count)?;
+        let num_candidates = dev
+            .max_number_candidates
+            .unwrap_or(semantic_search_config.max_number_candidates);
+        validate_count(
             count,
             semantic_search_config.max_number_documents,
-            semantic_search_config.default_number_documents,
-            semantic_search_config.max_number_candidates,
+            num_candidates,
         )?;
         let personalize = personalize
             .map(|personalize| personalize.validate(config.as_ref(), warnings))
             .transpose()?;
+        let dev_hybrid_search = dev.hybrid;
         let filter = Filter::insert_published_after(filter, published_after);
         let is_deprecated = published_after.is_some();
 
         Ok(SemanticSearchRequest {
             document,
             count,
+            num_candidates,
             personalize,
             enable_hybrid_search,
-            dev,
+            dev_hybrid_search,
```

Comment:
:wrench:  I would not strip out the dev options here as it means doing way more changes when adding any other dev options. Especially given what dev options are for this seems like a bad idea.

Reply:
the issue i see is that without splitting the dev option into its components, we carry around parts of the dev options into the route block, for example `dev.max_number_candidates`, which are not supposed to be used there since it was already used in the request validation where it got updated.

so i guess it's an argument of boilerplate vs logic bugs, where i tend to go with the former. wdyt?

Reply:
in the end it's at the messy boarder between input parsing, validation and pre-processing so I guess we can keep it the way you changed it to